leshan-client-core/src/main/java/org/eclipse/leshan/client/object/Oscore.java

```diff
@@ -64,8 +64,9 @@ public Oscore() {
     /**
      * Default constructor.
      */
-    public Oscore(String masterSecret, String senderId, String recipientId, int aeadAlgorithm, int hkdfAlgorithm, String masterSalt) {
-        this.masterSecret = masterSecret;
+    public Oscore(int instanceId, String masterSecret, String senderId, String recipientId, int aeadAlgorithm, int hkdfAlgorithm, String masterSalt) {
+        this.setId(instanceId);
```

Comment:
clearly not a big deal but you can use `super(id)`

Reply:
True, I will change that as part of the next commit.